### PR TITLE
[TargetActivity] allow to provide a custom comparator for target acti…

### PR DESCRIPTION
**Pull request by tbarthel-fr**
…vities.

### CODE REVIEW

File: library/src/main/java/fr/tvbarthel/intentshare/IntentShare.java [hunk 1]
```diff
@@ -171,6 +179,23 @@ public IntentShare iconLoader(@NonNull IconLoader iconLoader) {
...
+     * bring the magic.
+     *
+     * @param comparatorProvider comparator used to sort the TargetActivities displayed to the user.
+     *                           Will override the default sorting by recentness.
+     * @return current {@link IntentShare} for method chaining.
+     */
+    public IntentShare comparatorProvider(@NonNull TargetActivityComparatorProvider comparatorProvider) {
+        this.comparatorProvider = comparatorProvider;
+        return this;
+    }
```
Comment (vbarthel-fr) [inline library/src/main/java/fr/tvbarthel/intentshare/IntentShare.java:199]: I think that the `@NonNull` annotation should not be used here.

According to the documentation the `@NonNull` annotation `denotes that a parameter, field or method return value can never be null.` 

From my understanding of the `comparatorProvider` methods, you do not control whether or not the comparatorProvider is null, the client of your API does. If you do not want to support null parameter you should say so in the documentation and throw an NPE if comparatorProvider happens to be null.

At the moment, if by mistake I call `IntentShare.comparatorProvider()` from my code with a null reference, the code will only yield an NPE in the `TargetChooserActivity.onCreate()` methods (in the library code), leaving no clue of what went wrong. I would prefer to have an NPE thrown directly from `IntentShare.comparatorProvider()` notifying me of _my_ error, not _yours_ :)

File: library/src/main/java/fr/tvbarthel/intentshare/TargetActivity.java [hunk 2]
```diff
@@ -141,11 +125,20 @@ public boolean isMailClient() {
     }
 
     /**
+     * Return a timestamp of the last selection inside the sharing dialog from you application.
+     *
+     * @return timestamp of the last selection or 0 if the target activity has never been selected by the user.
+     */
+    public long getLastSelection() {
+        return lastSelection;
+    }
```
Comment (vbarthel-fr) [inline library/src/main/java/fr/tvbarthel/intentshare/TargetActivity.java:135]: Is `getLastSelection` a unix time stamp? What is the unit of this time stamp?

**Comment by tbarthel-fr** [comment]
add : 
- NPE for null custom comparator provider;
- Comment regarding timestamp format exposed for last selection;

**Comment by vbarthel-fr** [comment]
lgtm :+1:

**Pull request merged**